syntax = "proto3";

message RpcEnvelope {
  string method = 1;
  uint32 request_id = 2;
  bytes payload = 3;
}

message RpcResponse {
  uint32 request_id = 1;
  bytes payload = 2;
  string status = 3; // e.g., "OK", "ERROR"
}
